### PR TITLE
adding basic stats UI for reading log feature

### PR DESCRIPTION
**Pull request by mekarpeles**
Addresses issue #724 Reading Log Stats Page

### CODE REVIEW

**Comment by mekarpeles** [comment]
When possible, would love to deploy this. It adds an endpoint https://dev.openlibrary.org/stats/readinglog

File: openlibrary/templates/stats/readinglog.html [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      </tbody>
+    </table>
+  </div>
+
+  <div class="admin-section">
+    <h3>Most Wanted Books (All Time)</h3>
+    <div class="mybooks-list">
+      <ul class="list-books" id="siteSearch">
+        $for book in stats['leaderboard']['most_wanted']: 
+            $:macros.SearchResultsWork(book['work'], decorations="Added %s times" % book['cnt'])
```
Comment (labanlabs) [inline openlibrary/templates/stats/readinglog.html]: If you're going to string interpolate, please get the pluralization right, i.e. "1 times".

File: openlibrary/templates/stats/readinglog.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      </ul>
+    </div>
+
+    <h2>Most Logged Books</h2>
+    <h3>Most Read Books (All Time)</h3>
+    <div class="mybooks-list">
+      <ul class="list-books" id="siteSearch">
+        $for book in stats['leaderboard']['most_read']:
+            $:macros.SearchResultsWork(book['work'], decorations="Added %s times" % book['cnt'])
```
Comment (labanlabs) [inline openlibrary/templates/stats/readinglog.html]: If you're going to string interpolate, please get the pluralization right, i.e. "1 times".

File: openlibrary/views/loanstats.py [hunk 3]
```diff
@@ -96,3 +98,47 @@ def is_enabled(self):
     def GET(self):
         stats = WLStats()
         return app.render_template("stats/waitinglists.html", stats)
+
+class readinglog_stats(app.view):
+    path = "/stats/readinglog"
+
+    def GET(self):
+        ONE_MONTH_DATE = datetime.date.today() - datetime.timedelta(days=28)
```
Comment (labanlabs) [inline openlibrary/views/loanstats.py]: I think we need to be able to specify the date span.

File: openlibrary/views/loanstats.py [hunk 4]
```diff
@@ -96,3 +98,47 @@ def is_enabled(self):
...
+                },
+                'leaderboard': {
+                    'most_read': most_read,
+                    'most_wanted': most_wanted
+                }
+            }
+        stats =  cache.memcache_memoize(
+            readinglog_stats, 'stats.readlinglog_stats', timeout=cache.HOUR)(limit)
+
+        # Fetch works from solr and inject into leaderboard
```
Comment (labanlabs) [inline openlibrary/views/loanstats.py]: What's the reliability of these stats, given they're from solr?

Comment (mekarpeles) [reply openlibrary/views/loanstats.py]: @labanlabs the readinglog stats are going directly from postgres so they are accurate. They are tracked in a different way than most of our systems (intentionally). That said, the results are cached for an hour, so we expect up to an hour delay in the results.

File: openlibrary/templates/stats/readinglog.html (hunk 1 above)
Comment (mekarpeles) [reply openlibrary/templates/stats/readinglog.html]: Can do -- thanks for the suggestion!

File: openlibrary/views/loanstats.py (hunk 3 above)
Comment (mekarpeles) [reply openlibrary/views/loanstats.py]: This is probably right -- at this point I'd prefer opening a separate issue and supporting it as its needed. The primary use case for this feature was/is internal weekly metrics updates (which we want the public to have access to as well). If anyone feels querying by date is a blocker, I'm happy to spend time reviewing a pull request which addresses that.

**Comment by mekarpeles** [comment]
@labanlabs, apologies! I switched branches for testing purpose. I'll update here as soon as its switched back (should be an hour or so). Sorry for the inconvenience and thank you so much for providing a code review! Really appreciate it.

**Comment by mekarpeles** [comment]
dev is currently back on this branch and the suggested change re: time v. times has been made

**Pull request merged**